src/System.IO.Compression/src/System/IO/Compression/DeflateStream.cs

```diff
@@ -531,6 +532,7 @@ protected override void Dispose(bool disposing)
                     }
                     finally
                     {
+                        ArrayPool<byte>.Shared.Return(_buffer);
```

Comment:
Should we be setting clearArray to true?  We're returning buffers that could potentially have been storing data that shouldn't be exposed later to others using the pool.